pkg/services/ngalert/store/image.go

```diff
@@ -75,6 +79,19 @@ func (st DBstore) GetImageByURL(ctx context.Context, url string) (*models.Image,
 	return &image, nil
 }
 
+func (st DBstore) URLExists(ctx context.Context, url string) (bool, error) {
+	var exists bool
+	err := st.SQLStore.WithDbSession(ctx, func(sess *db.Session) error {
+		ok, err := sess.Table("alert_image").Where("url = ? AND expires_at > ?", url, TimeNow().UTC()).Exist()
```

Comment:
Neither the `url` nor the `expires_at` field got a index. That might be something you want to consider, otherwise, it could slow down this operation. If not, please add a comment about why you think this is not necessary.